gluten-core/src/main/scala/io/glutenproject/extension/RemoveKnownFloatingPointNormalized.scala

```diff
@@ -0,0 +1,80 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.extension
+
+import io.glutenproject.extension.columnar.TransformHints
+import io.glutenproject.utils.PhysicalPlanSelector
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.optimizer.NormalizeNaNAndZero
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.aggregate.{BaseAggregateExec, HashAggregateExec, ObjectHashAggregateExec, SortAggregateExec}
+
+/**
+ * [[KnownFloatingPointNormalized]] in Gluten is useless. When converting plans, it directly takes
+ * its child. We can reverse the KnownFloatingPointNormalized added by
+ * NormalizeFloatingNumbers.normalize in Gluten without affecting the results. Otherwise, pulling
+ * out pre-project in the logical plan is not sufficient. The KnownFloatingPointNormalized will be
+ * added to the groupingExpressions of the aggregate in the physical plan.
+ */
+case class RemoveKnownFloatingPointNormalized(session: SparkSession) extends Rule[SparkPlan] {
+
+  private def normalizeExpressionExists(expressions: Seq[Expression]): Boolean = {
+    expressions.exists(_.find(_.isInstanceOf[KnownFloatingPointNormalized]).isDefined)
+  }
+
+  /**
+   * If the groupingExpressions of the aggregate are expressions, an Alias will be added on the
+   * outside. After removing the KnownFloatingPointNormalized, it is necessary to remove this
+   * redundant Alias.
+   */
+  private def removeRedundantAlias(expression: Expression): Expression = expression match {
+    case _ @Alias(child: Attribute, _) => child
+    case other => other
+  }
+
+  override def apply(plan: SparkPlan): SparkPlan = PhysicalPlanSelector.maybe(session, plan) {
+    plan.transform {
+      case agg: BaseAggregateExec
+          if normalizeExpressionExists(agg.groupingExpressions) &&
+            !TransformHints.isAlreadyTagged(plan) =>
+        val newGroupingExprs = agg.groupingExpressions.toIndexedSeq.map {
+          expr =>
+            val transformedExpr = expr.transform {
+              case _ @KnownFloatingPointNormalized(_ @NormalizeNaNAndZero(child)) => child
```

Comment:
Can we remove `KnownFloatingPointNormalized` and `NormalizeNaNAndZero` in `ExpressionConverter` ?

Reply:
Yes. Removed.

Reply:
@ulysses-you We cannot remove, both window and join still use them. They are handled in the logical plan rule `NormalizeFloatingNumbers`, so we don't need to eliminate it. We only need to remove the `NormalizeFloatingNumbers` in agg. Although we could also include join and window in `RemoveKnownFloatingPointNormalized`, I think it's unnecessary. In the future, when Spark handles agg within the logical plan as well, we can remove `RemoveKnownFloatingPointNormalized` in Gluten.

Reply:
I see, after we pull out join/window, the NormalizeNaNAndZero will be moved to project and we skip tranform it at the end